### PR TITLE
Error when contract call via Contract instance with account in Node.

### PR DESCRIPTION
**Pull request by jimni1222**
## Proposed changes

Use Legacy transaction type when call or deploy the contract via Contract instance with account in Node.
And also to return new Contract instance as a return value of deploy, modify the condition check to decide 'isContractDeployment'.

## Types of changes

Please put an x in the boxes related to your change.

- [x] Bugfix
- [ ] New feature or enhancement
- [ ] Others

## Checklist

*Put an x in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code.*

- [x] I have read the [CONTRIBUTING GUIDELINES](https://github.com/klaytn/caver-js/blob/master/CONTRIBUTING.md) doc
- [x] I have signed the [CLA](https://cla-assistant.io/klaytn/caver-js)
- [x] Lint and unit tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

## Related issues

close https://github.com/klaytn/caver-js/issues/34

## Further comments

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: packages/caver-core-method/src/index.js [hunk 1]
```diff
@@ -107,7 +107,7 @@ function createFunction(requestManager, accounts) {
   // call is directly used for rpc calling,
   // ex) 'klay_sendTransaction'
   func.call = this.call
-
+
```
Comment (kjhman21) [inline packages/caver-core-method/src/index.js]: Please remove unnecessary changes.

File: packages/caver-core-method/src/index.js [hunk 2]
```diff
@@ -285,7 +285,7 @@ const buildSendTxCallbackFunc = (defer, method, payload, isSendTx) => (err, resu
 
   // fire callback
   if (payload.callback) payload.callback(null, result)
-
+
```
Comment (kjhman21) [inline packages/caver-core-method/src/index.js]: Please remove unnecessary changes.

File: packages/caver-core-method/src/index.js [hunk 3]
```diff
@@ -309,7 +309,7 @@ const buildSendRequestFunc = (defer, sendSignedTx, sendTxCallback) => (payload,
       switch (payload.method) {
         case 'klay_sendTransaction': {
           var tx = payload.params[0]
-          
+
```
Comment (kjhman21) [inline packages/caver-core-method/src/index.js]: Please remove unnecessary changes.

File: packages/caver-klay/caver-klay-contract/src/index.js [hunk 4]
```diff
@@ -1039,7 +1039,7 @@ Contract.prototype._executeMethod = function _executeMethod(){
                 if (args.options.type !== 'SMART_CONTRACT_EXECUTION' && args.options.type !== 'SMART_CONTRACT_DEPLOY') {
                     throw new Error('Unsupported transaction type. Please use SMART_CONTRACT_EXECUTION or SMART_CONTRACT_DEPLOY.')
                 } 
-
+
```
Comment (kjhman21) [inline packages/caver-klay/caver-klay-contract/src/index.js]: Please remove unnecessary changes.

File: packages/caver-core-method/src/index.js (hunk 1 above)
Comment (jimni1222) [reply packages/caver-core-method/src/index.js]: Okay, i removed unnecessary changes :)

File: packages/caver-core-method/src/index.js (hunk 2 above)
Comment (jimni1222) [reply packages/caver-core-method/src/index.js]: Okay, i removed unnecessary changes :)

File: packages/caver-core-method/src/index.js (hunk 3 above)
Comment (jimni1222) [reply packages/caver-core-method/src/index.js]: Okay, i removed unnecessary changes :)

File: packages/caver-klay/caver-klay-contract/src/index.js (hunk 4 above)
Comment (jimni1222) [reply packages/caver-klay/caver-klay-contract/src/index.js]: Okay, i removed unnecessary changes :)
